go/libraries/doltcore/sqle/remotesrv.go

```diff
@@ -60,9 +60,14 @@ func (s remotesrvStore) Get(path, nbfVerStr string) (remotesrv.RemoteSrvStore, e
 	return rss, nil
 }
 
-func RemoteSrvServerArgs(ctx *sql.Context, args remotesrv.ServerArgs) remotesrv.ServerArgs {
+func RemoteSrvServerArgs(ctx *sql.Context, args remotesrv.ServerArgs, userAuth *remotesrv.UserAuth) remotesrv.ServerArgs {
```

Comment:
Nit: I think I would encourage something like:

```
func WithUserPasswordAuth(args remotesrv.ServerArgs, userAuth *remotesrv.UserAuth) remotesrv.ServerArgs
```

and leaving this function alone, since it doesn't touch `args.Options` as is.

I think with that structure, `ServerArgs.ServerInterceptor` could be left off. The server interceptor struct could be public and the `WithUserPasswordAuth` function could just construct an interceptor with an appropriate value based on the passed in `UserAuth` pointer.

Feel free to push back if this makes it noticeably worse, but it seems right to me.